tests/jax/models/test_simple_nn.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Sequence
+
+import jax
+import jax.numpy as jnp
+import pytest
+from flax import nnx
+from infra import BaseModelTester, ComparisonConfig, Workload
+
+
+class SimpleNN(nnx.Module):
+    def __call__(
+        self, act: jax.Array, w0: jax.Array, b0: jax.Array, w1: jax.Array, b1: jax.Array
+    ) -> jax.Array:
+        x = jnp.matmul(act, w0) + b0
+        x = jnp.matmul(x, w1) + b1
+        return x
+
+
+class SimpleNNTester(BaseModelTester):
+    @staticmethod
+    def _get_model() -> nnx.Module:
+        return SimpleNN()
+
+    @staticmethod
+    def _get_model_inputs() -> Sequence[jax.Array]:
+        act_shape, w0_shape, b0_shape, w1_shape, b1_shape = (
+            (32, 784),
+            (784, 128),
+            (1, 128),
+            (128, 128),
+            (1, 128),
+        )
+
+        act = jax.numpy.ones(act_shape)
+        w0 = jax.numpy.ones(w0_shape)
+        b0 = jax.numpy.ones(b0_shape)
+        w1 = jax.numpy.ones(w1_shape)
+        b1 = jax.numpy.zeros(b1_shape)
+
+        return [act, w0, b0, w1, b1]
+
+    # @override
+    @staticmethod
+    def _get_model_as_workload() -> Workload:
+        return Workload(SimpleNNTester._get_model(), SimpleNNTester._get_model_inputs())
+
+    # @override
+    @staticmethod
+    def _get_model_forward_pass_method_name() -> str:
+        return "__call__"
+
+
+@pytest.fixture
+def comparison_config() -> ComparisonConfig:
+    config = ComparisonConfig()
+    config.disable_all()
+    config.pcc.enable()
```

Comment:
Are model tests failing other metrics?

Reply:
Atol is 64.  PCC and allclose pass.

Reply:
Remember that I already ran into a similar situation https://github.com/tenstorrent/tt-metal/issues/15130

Reply:
That's wild that atol is failing with such a high difference, while allclose passes. Allclose is calculated as:
```
absolute(a - b) <= (atol + rtol * absolute(b))
```

which tells me that the only way for this to happen is either:
1. Output activations get insanely high (>=6400) so that `64 <= 0.01 + 0.01 * activation`
2. We are running comparison functions on device instead of cpu, which might cause incorrect calculations

Looking at the way you initialized `ExampleModel` weights, it seems like the 1st case is the root cause. Inputs are randomly initialized in `[0, 1]` range, first matmul gives `784*[0, 1] + 1 = [1, 785]`. Second matmul gives `128 * [1, 785] + 0 = [128, 100480]`. You should initialize weights with random values in some smaller range like `[-0.01, 0.01]`, biases too. Then you will be able to enable atol comparison too.

Additionally, I think we should make sure that the comparison functions are running on CPU, I'll add a separate comment for that in `BaseTester`.

Reply:
You are right.

Expanded `random_tensor` util function with `minval` and `maxval` and redefined biases and weights in `ExampleModel` as

```python
self.w0 = random_tensor(w0_shape, minval=-0.01, maxval=0.01)
self.w1 = random_tensor(w1_shape, minval=-0.01, maxval=0.01)
self.b0 = random_tensor(b0_shape, minval=-0.01, maxval=0.01)
self.b1 = random_tensor(b1_shape, minval=-0.01, maxval=0.01)
```

Now it passes everything. Removed `ComparisonConfig` fixture since it is unneeded.